Update GTest fort compatibility with GCC11

|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.3.0 | GTest 1.8 | Manager/Agent | Packages/Sources | OS version |

## Description
This issue aims to update the GTest version with the idea of ​​absorbing the PR fix: https://github.com/google/googletest/pull/3024


Actual:
```
[ 12%] Building CXX object googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.o
In file included from /wazuh/src/external/googletest/googletest/src/gtest-all.cc:42:
/wazuh/src/external/googletest/googletest/src/gtest-death-test.cc: In function 'bool testing::internal::StackGrowsDown()':
/wazuh/src/external/googletest/googletest/src/gtest-death-test.cc:1301:24: error: 'dummy' may be used uninitialized [-Werror=maybe-uninitialized]
 1301 |   StackLowerThanAddress(&dummy, &result);
      |   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
/wazuh/src/external/googletest/googletest/src/gtest-death-test.cc:1290:13: note: by argument 1 of type 'const void*' to 'void testing::internal::StackLowerThanAddress(const void*, bool*)' declared here
 1290 | static void StackLowerThanAddress(const void* ptr, bool* result) {
      |             ^~~~~~~~~~~~~~~~~~~~~
/wazuh/src/external/googletest/googletest/src/gtest-death-test.cc:1299:7: note: 'dummy' declared here
 1299 |   int dummy;
      |       ^~~~~
cc1plus: all warnings being treated as errors
make[3]: *** [googletest/CMakeFiles/gtest.dir/build.make:82: googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.o] Error 1
make[3]: Leaving directory '/wazuh/src/external/googletest/build'
make[2]: *** [CMakeFiles/Makefile2:219: googletest/CMakeFiles/gtest.dir/all] Error 2
make[2]: Leaving directory '/wazuh/src/external/googletest/build'
make[1]: *** [Makefile:160: all] Error 2
make[1]: Leaving directory '/wazuh/src/external/googletest/build'
make: *** [Makefile:2038: build_gtest] Error 2
```

Expected:
successful compilation of unit tests.


## DoD
- [x] Create package with `tar -zcf googletest.tar.gz --owner=0 --group=0 googletest` of https://github.com/google/googletest/archive/refs/tags/release-1.11.0.tar.gz
- [x] Update S3 package -> version 15 of deps.
- [x] Apply source fix
```
[ 66%] Building CXX object tests/sysInfo/CMakeFiles/sysinfo_unit_test.dir/sysInfo_test.cpp.o
/home/dwordcito/Development/wazuh/src/data_provider/tests/sysInfo/sysInfo_test.cpp: In member function ‘virtual void SysInfoTest_packages_cb_Test::TestBody()’:
/home/dwordcito/Development/wazuh/src/data_provider/tests/sysInfo/sysInfo_test.cpp:167:48: error: ‘DoAll’ was not declared in this scope; did you mean ‘testing::DoAll’?
  167 |     EXPECT_CALL(info, getPackages(_)).WillOnce(DoAll(
      |                                                ^~~~~
      |                                                testing::DoAll
In file included from /home/dwordcito/Development/wazuh/src/external/googletest/googlemock/include/gmock/gmock.h:59,
                 from /home/dwordcito/Development/wazuh/src/data_provider/tests/sysInfo/sysInfo_test.h:15,
                 from /home/dwordcito/Development/wazuh/src/data_provider/tests/sysInfo/sysInfo_test.cpp:11:
/home/dwordcito/Development/wazuh/src/external/googletest/googlemock/include/gmock/gmock-actions.h:1204:61: note: ‘testing::DoAll’ declared here
 1204 | internal::DoAllAction<typename std::decay<Action>::type...> DoAll(
      |                                                             ^~~~~
/home/dwordcito/Development/wazuh/src/data_provider/tests/sysInfo/sysInfo_test.cpp: In member function ‘virtual void SysInfoTest_processes_cb_Test::TestBody()’:
/home/dwordcito/Development/wazuh/src/data_provider/tests/sysInfo/sysInfo_test.cpp:205:53: error: ‘DoAll’ was not declared in this scope; did you mean ‘testing::DoAll’?
  205 |     EXPECT_CALL(info, getProcessesInfo(_)).WillOnce(DoAll(
      |                                                     ^~~~~
      |                                                     testing::DoAll
In file included from /home/dwordcito/Development/wazuh/src/external/googletest/googlemock/include/gmock/gmock.h:59,
                 from /home/dwordcito/Development/wazuh/src/data_provider/tests/sysInfo/sysInfo_test.h:15,
                 from /home/dwordcito/Development/wazuh/src/data_provider/tests/sysInfo/sysInfo_test.cpp:11:
/home/dwordcito/Development/wazuh/src/external/googletest/googlemock/include/gmock/gmock-actions.h:1204:61: note: ‘testing::DoAll’ declared here
 1204 | internal::DoAllAction<typename std::decay<Action>::type...> DoAll(
      |                                                             ^~~~~
make[4]: *** [tests/sysInfo/CMakeFiles/sysinfo_unit_test.dir/build.make:104: tests/sysInfo/CMakeFiles/sysinfo_unit_test.dir/sysInfo_test.cpp.o] Error 1
make[4]: Leaving directory '/home/dwordcito/Development/wazuh/src/data_provider/build'
make[3]: *** [CMakeFiles/Makefile2:291: tests/sysInfo/CMakeFiles/sysinfo_unit_test.dir/all] Error 2
make[3]: Leaving directory '/home/dwordcito/Development/wazuh/src/data_provider/build'
make[2]: *** [Makefile:101: all] Error 2
make[2]: Leaving directory '/home/dwordcito/Development/wazuh/src/data_provider/build'
make[1]: *** [Makefile:1445: build_sysinfo] Error 2
make[1]: Leaving directory '/home/dwordcito/Development/wazuh/src'
make: *** [Makefile:720: agent] Error 2
```

--- a/src/data_provider/tests/sysInfo/sysInfo_test.cpp
+++ b/src/data_provider/tests/sysInfo/sysInfo_test.cpp
@@ -30,6 +30,7 @@ auto PACKAGES_EXPECTED
 
 using ::testing::_;
 using ::testing::Return;
+using ::testing::DoAll;
 
 std::string SysInfo::getSerialNumber() const
 {

